devito/finite_differences/finite_difference.py

```diff
@@ -235,26 +219,35 @@ def generic_derivative(expr, dim, fd_order, deriv_order, stagger=None, symbolic=
     expr-like
         ``deriv-order`` derivative of ``expr``.
     """
-    diff = dim.spacing
-
+    # First order derivative with 2nd order FD is highly non-recommended so taking
+    # first order fd that is a lot better
+    if deriv_order == 1 and fd_order == 2 and not symbolic:
```

Comment:
Not sure.... raise warning? (@rhodrin )

Reply:
reminder

Reply:
Sorry for the delay 😬

Yeah, should raise a warming is some user input is being changed I'd say.

Reply:
User input is not changed. First order derivative don't have a second order approx, only a 1st and 3rd order, so for order 2 only possible one is 1st order. Then you have three choices which are `[x-hx, 0,  h+hx], [0, x, x+hx], [0, x, x-hx]`. This one just choose the ` [0, x, x+hx]` one that is lot better than the `[x-hx, 0,  h+hx]` the should usually be avoided at all cost because it skips half the grid.

Reply:
Agreed. Nobody here is saying not to do it the current way; we were just debating about whether a warning should be raised since the user requests one thing, but gets a different one